homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -147,80 +147,85 @@ def __init__(
         self._weekday = weekday
         self._time = departuretime
         self._state: TrainStop | None = None
-        self._delay_in_minutes = None
+        self._delay_in_minutes: timedelta | None = None
         self._timezone = get_time_zone("Europe/Stockholm")
-        self._attr_extra_state_attributes = {
-            ATTR_DEPARTURE_STATE: None,
-            ATTR_CANCELED: None,
-            ATTR_DELAY_TIME: None,
-            ATTR_PLANNED_TIME: None,
-            ATTR_ESTIMATED_TIME: None,
-            ATTR_ACTUAL_TIME: None,
-            ATTR_OTHER_INFORMATION: None,
-            ATTR_DEVIATIONS: None,
-        }
+        self._attr_extra_state_attributes = {}
 
     async def async_update(self) -> None:
         """Retrieve latest state."""
-        when = datetime.now()
-        if self._time:
-            departure_day = next_departuredate(self._weekday)
-            when = datetime.combine(departure_day, self._time).astimezone(
-                self._timezone
-            )
         try:
             if self._time:
+                departure_day = next_departuredate(self._weekday)
+                when = datetime.combine(departure_day, self._time).astimezone(
+                    self._timezone
+                )
```

Comment:
You shouldn't have moved this inside the try...except. Only the code that might raise the error should be inside.